presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -198,10 +204,88 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     {
         CassandraTableHandle handle = checkType(table, CassandraTableHandle.class, "table");
         CassandraPartitionResult result = partitionManager.getPartitions(handle, constraint.getSummary());
-        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions()));
+        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions(), buildClusteringPredicate(schemaProvider.getTable(handle), constraint)));
         return ImmutableList.of(new ConnectorTableLayoutResult(layout, result.getUnenforcedConstraint()));
     }
 
+    private static Optional<String> buildClusteringPredicate(CassandraTable cassandraTable, Constraint<ColumnHandle> constraint)
```

Comment:
It would be nice to add unit tests for this method. Maybe you can extract this algorithm to a separate class, and then unit-test it.